src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -0,0 +1,1349 @@
+        // content, which is unexpected.
+        const auto htmlData = formats == nullptr || WI_IsFlagSet(formats.Value(), CopyFormat::HTML) ?
+                                  TextBuffer::GenHTML(bufferData,
+                                                      _actualFont.GetUnscaledSize().Y,
+                                                      _actualFont.GetFaceName(),
+                                                      til::color{ _settings.DefaultBackground() }) :
+                                  "";
+
+        // convert to RTF format
+        const auto rtfData = formats == nullptr || WI_IsFlagSet(formats.Value(), CopyFormat::RTF) ?
+                                 TextBuffer::GenRTF(bufferData,
+                                                    _actualFont.GetUnscaledSize().Y,
+                                                    _actualFont.GetFaceName(),
+                                                    til::color{ _settings.DefaultBackground() }) :
+                                 "";
+
+        if (!_settings.CopyOnSelect())
+        {
+            _terminal->ClearSelection();
+            _renderer->TriggerSelection();
+        }
+
+        // send data up for clipboard
+        auto copyArgs = winrt::make_self<CopyToClipboardEventArgs>(winrt::hstring(textData),
+                                                                   winrt::to_hstring(htmlData),
+                                                                   winrt::to_hstring(rtfData),
+                                                                   formats);
+        _CopyToClipboardHandlers(*this, *copyArgs);
+        return true;
+    }
+
+    // Method Description:
+    // - Pre-process text pasted (presumably from the clipboard)
+    //   before sending it over the terminal's connection.
+    void ControlCore::PasteText(const winrt::hstring& hstr)
+    {
+        _terminal->WritePastedText(hstr);
+        _terminal->ClearSelection();
+        _terminal->TrySnapOnInput();
+    }
+
+    FontInfo ControlCore::GetFont() const
+    {
+        return _actualFont;
+    }
+
+    til::size ControlCore::FontSizeInDips() const
+    {
+        const til::size fontSize{ GetFont().GetSize() };
+        return fontSize.scale(til::math::rounding, 1.0f / ::base::saturated_cast<float>(_compositionScaleX));
+    }
+
+    TerminalConnection::ConnectionState ControlCore::ConnectionState() const
+    {
+        return _connection.State();
+    }
+
+    hstring ControlCore::Title()
+    {
+        hstring hstr{ _terminal->GetConsoleTitle() };
+        return hstr;
+    }
+
+    hstring ControlCore::WorkingDirectory() const
+    {
+        hstring hstr{ _terminal->GetWorkingDirectory() };
+        return hstr;
+    }
+
+    bool ControlCore::BracketedPasteEnabled() const noexcept
+    {
+        return _terminal->IsXtermBracketedPasteModeEnabled();
+    }
+
+    Windows::Foundation::IReference<winrt::Windows::UI::Color> ControlCore::TabColor() noexcept

```

Comment:
why is TabColor something that ....

just kidding 😁

Reply:
Maybe? This one's weird and I didn't want to mess with it, so I'm gonna leave #wontfix